mw-webapp/src/component/viewSwitcher/ViewSwitcher.tsx

```diff
@@ -0,0 +1,95 @@
+import clsx from "clsx";
+import {HorizontalContainer} from "src/component/horizontalContainer/HorizontalContainer";
+import {Icon, IconProps, IconSize} from "src/component/icon/Icon";
+import {PositionTooltip} from "src/component/tooltip/PositionTooltip";
+import {Tooltip, TooltipProps} from "src/component/tooltip/Tooltip";
+import {View} from "src/utils/LocalStorageWorker";
+import styles from "src/component/viewSwitcher/ViewSwitcher.module.scss";
+
+/**
+ * View option
+ */
+export interface ViewOption {
+
+  /**
+   * View types
+   */
+  view: View;
+
+  /**
+   * Tooltip's content
+   */
+  tooltipContent: TooltipProps["content"];
+
+  /**
+   * Icon name
+   */
+  iconName: IconProps["name"];
```

Comment:
Maybe better to use keyof typeof IconDictionary?

Reply:
It's a philosophical question whether viewSwitcher needs to relate on IconProps or not. On the one hand, viewSwitcher uses IconProps and logically determines the type of the field. But at the same time, we “link” these two components directly + readability decreased (2 jumps instead of one, the ability to “jump through the search” without an IDE is difficult). And most important -  it’s good to make components as independent as possible at the interface level. For example, it now seems that creating a viewSwitcher requires that the IconProps have a name field, which from a design point of view is an awkward statement that will be confusing and raise "why" questions. My personal opinion is that there is no need to inherit types if we don't have good reasons, hierarchies are really  difficult to maintain if they do not have systems. That's why some people hate OOP - with no good system the web of relation is a mess

Reply:
Replaced with keyof typeof IconDictionary